list.lisp

```diff
@@ -0,0 +1,116 @@
+;;;; -*- Mode: Lisp; indent-tabs-mode: nil -*-
+
+(in-package :split-sequence)
+
+(declaim (inline
+          collect-until count-while
+          split-list split-list-if split-list-if-not
+          split-list-from-end split-list-from-start split-list-internal))
+
+(declaim (ftype (function (&rest t) (values list unsigned-byte))
+                split-list split-list-if split-list-if-not))
+
+(declaim (ftype (function (function list unsigned-byte (or null unsigned-byte) (or null unsigned-byte)
+                                    boolean)
+                          (values list unsigned-byte))
+                split-list-from-start split-list-from-end split-list-internal))
+
+(defun collect-until (predicate list end)
+  "Collect elements from `list` until one that satisfies `predicate` is found.
```

Comment:
```suggestion
  "Collect elements from LIST until one that satisfies PREDICATE is found.
```

It looks like you're going with UPPERCASE to denote symbol names.  That's fine, but we should be consistent and fix my markdown-formatted docstrings.